src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowAction.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.rest;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.common.FlowFrameworkFeatureEnabledSetting;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.transport.GetWorkflowAction;
+import org.opensearch.flowframework.transport.WorkflowRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_ID;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_URI;
+import static org.opensearch.flowframework.common.FlowFrameworkFeatureEnabledSetting.FLOW_FRAMEWORK_ENABLED;
+
+/**
+ * Rest Action to facilitate requests to get a workflow status
+ */
+public class RestGetWorkflowAction extends BaseRestHandler {
```

Comment:
Perhaps lets change this rest action name to `RestGetWorkflowStateAction`, since its returning an object of type `WorkflowState` instead of `Workflow`

Reply:
I was trying to follow the same pattern as AD here where we have a GET action for everything that has the pattern of `_plugins/ad/detectors/<id>` and then there are different parameters like adding `_profile`. So when we do get workflow after, its the same place, we just can choose whether to add the _status after the id. I can change it too getworkflowstatus for now though or we can keep having the separate actions for get status vs workflow

Reply:
It's fine to keep it as `RestGetWorkflowAction` and let's add `_status` param in the request then.

Reply:
Agreed